### ISSUE TITLE
Segmentation Fault after any async work reading headers

### ISSUE THREAD

**Issue by gornostay25** [oven-sh/bun#489]
I try to run SvelteKit build with custom adapter, but it returns segmentation fault
![image](https://user-images.githubusercontent.com/36516130/178103235-d0a16699-d6e3-4a55-a34a-c5c4742d8ee0.png)

**Comment by ItzDerock** [comment]
Ran into this issue too, doing anything that references the headers causes a crash
```ts
export default {
    port: 3000,
    async fetch(request: Request) {
        const body = await request.text();

        request.headers;

        return new Response("hello world");
    }
}
```

but after trying some more things, I found that creating and resolving your own promise works:  
`await new Promise(r => setTimeout(r, 500))` will not crash.

but calling things like `await request.blob()` or anything to do with the body of the request will cause it to crash.

**Comment by gornostay25** [comment]
https://github.com/gornostay25/svelte-adapter-bun/blob/7245e0ae3cf79df622d14d8b4f9f7ca032061502/src/handler.js#L16

@ItzDerock Thanks 👍

**Comment by Jarred-Sumner** [comment]
Fixed in Bun v0.1.8 I think

Definitely fixed now though
